library/analytics/src/test/java/org/cru/godtools/analytics/appsflyer/AppsFlyerAnalyticsServiceTest.kt

```diff
@@ -0,0 +1,80 @@
+package org.cru.godtools.analytics.appsflyer
+
+import android.app.Activity
+import android.app.Application
+import android.content.Intent
+import android.net.Uri
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.appsflyer.AppsFlyerLibCore
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.anyOrNull
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.isNull
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.never
+import com.nhaarman.mockitokotlin2.same
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import org.greenrobot.eventbus.EventBus
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class AppsFlyerAnalyticsServiceTest {
+    private lateinit var application: Application
+    private lateinit var activity: Activity
+    private lateinit var eventBus: EventBus
+    private lateinit var deepLinkResolver: AppsFlyerDeepLinkResolver
+
+    private lateinit var analyticsService: AppsFlyerAnalyticsService
+
+    @Before
+    fun setupMocks() {
+        AppsFlyerLibCore.instance = mock()
+        application = mock()
+        activity = mock()
+        eventBus = mock()
+        deepLinkResolver = mock()
+
+        analyticsService = AppsFlyerAnalyticsService(application, eventBus, setOf(deepLinkResolver))
+    }
+
+    // region onAppOpenAttribution()
+    @Test
```

Comment:
Can you test any func as long as you add this ```@Test``` above it, or is there other requirements as well?

Reply:
the `@Test` annotation indicates that a function is a test entry point for JUnit 4. so every function/method that is annotated with `@Test` will be run by the unit test framework and will report success or any failures

Reply:
all tests live in a separate source set

Reply:
we talked about how source sets and unit testing work on android, so resolving this